docs/javascript/v2/advanced-features/session-metadata.mdx

```diff
@@ -0,0 +1,270 @@
+---
+title: Session Metadata
+nav: 5.04
+---
+
+Session metadata is a shared state that is accessible by everyone in the room. Think of it as a additional top-level data associated with a session. Session metadata can be used to achieve features like pinned text, spotlight(bringing a particular peer into a center stage for everyone in the room), etc.
+
+Session metadata is persisted throughout a session and is cleared once the last peer leaves the room.
+
+<Note>
+💡 Session Metadata vs Peer Metadata
+
+While peer metadata is associated with individual peers and each peer can have their own metadata, session metadata remains the same for every peer in the room.
+
+</Note>
+
+## SDK Actions
+
+### setSessionMetadata
+
+-   Use `setSessionMetadata` to update the value of the session metadata.
+-   Objects could be set as session metadata but the object should be JSON serializable.
+-   Note: whatever is passed to setSessionMetadata is set as session metadata as is. When using objects, if you want to update a particular key, you should update it before calling setSessionMetadata.
+
+```tsx
+// interface
+setSessionMetadata(metadata: any): Promise<void>
+
+// usage
+await hmsActions.setSessionMetadata('any-string-here');
+await hmsActions.setSessionMetadata({ hello: 'world' });
+```
+
+### populateSessionMetadata
```

Comment:
not clear what this is supposed to do. can you add a line on how a user is supposed to use it/when to use it?

i gathered that i need to call this before doing a `getSessionMetadata` call to sync with the server. is that correct?